### DISCUSSION TITLE
Lack of several features in self-host

### DISCUSSION THREAD

**Discussion by 0xlimon** [supabase/supabase#12151] [General]
There are more features in supabase cloud than self-host. such as logs..reports..providers..email templates and so on
Is there a part to enable these features or are they not in self-host?

**Comment by globemediaofficial** [comment]
@awminux I felt the same way when I first started using Supabase Self-Hosted but there is really more a lack of documentation than a lack of features. Assuming you mean OAuth providers by "providers", this is actually something that you can enable, just with variables inside `docker-compose.yml` instead of simple UI in the `Studio`, but the instructions for this are hard to find and don't even explain all of the steps. I have created an unofficial guide that goes more in-depth on some missing instructions. [Here is the link to the OAuth section of that guide](https://medium.com/@christopher.mathews/how-to-self-host-supabase-on-your-own-computer-full-guide-cceac9b740ea#a2af). As for the logs, reports, and email templates, I have not needed to use those, so I don't know about them.

**Comment by ruggi99** [comment]
Hi @awminux.
Sadly not all features from platform are supported in self-hosted version.
For example logs and report, as well as backups and so on are not included in Supabase itself, but only in the infrastructure which Supabase relies on.
For example there's nothing in this repository that collects logs and save them somewhere because it's an additional feature provided for cloud version.

**The main reason behind this is that you don't have access to the infrastructure of Supabase**.
So they have to provide a way to get data (logs, report) and so they provide a GUI to users, but are all additional features.

There are 2 more reason I think:
- Supabase's infrastructure has an independent service that collects all logs from all projects and store them somewhere, instead of having the same service for the single project (cost efficient).
- Because you have access to the "server" where you self-host Supabase you can also retrieve logs with docker compose, although there no GUI interface.

The same applies to reports.

Logs and report are not provided at all in self-hosted.

But for email templates and providers you have to "play" with environment variables in docker as said by @globemediaofficial .
Here is a list of all the env vars: https://github.com/supabase/gotrue (there are also env vars for email templates).

Summing up if you want to self-host you have to deal with docker and env vars.
A feature that may be lacking here is, for example, disable temporary new signups from the dashboard. If you want to achieve this you have to change `.env` file and recreate the container

**Comment by estubmo** [comment]
Additionally Email Template, Auth Providers, Realtime and Edge Function configuration are a ll missing from the Supabase Studio in the self-hosted version.

You can manage it through setting env variables, but it's quite an inconvenience. Especially email html templates being left out, so you have to manage self hosting of those.

**Reply by leiless** [reply]
> You can manage it through setting env variables

How can I do that? more specifically.

**Reply by estubmo** [reply]
For the email templates? Like this for example:
https://github.com/coollabsio/coolify/blob/7c1948ebd9ab831e9b267ae11e549c1abaee3e65/templates/compose/supabase.yaml#L941-L945

This docker-compose.yaml contains some stuff specific for coolify, but all of the mentioned services work.
